verifier/src/main/java/com/aws/jverify/verifier/compiler/frontend/InstrumentLower.java

```diff
@@ -0,0 +1,58 @@
+package com.aws.jverify.verifier.compiler.frontend;
+
+import com.sun.tools.javac.code.Symbol;
+import com.sun.tools.javac.util.List;
+import net.bytebuddy.agent.ByteBuddyAgent;
+import net.bytebuddy.agent.builder.AgentBuilder;
+import net.bytebuddy.description.modifier.Visibility;
+import net.bytebuddy.dynamic.Transformer;
+import net.bytebuddy.implementation.MethodDelegation;
+import java.lang.instrument.Instrumentation;
+
+import static net.bytebuddy.matcher.ElementMatchers.*;
+
+/**
+ * We override methods in Lower.java that create VarSymbol s
+ * so that they will not erase the assigned type.
+ * <p>
+ * We can not stop doing type erasure in Lower.java in general,
+ * because it depends on erasure for type comparisons and lookups.
+ */
+public class InstrumentLower {
+
+    public static void installModification() {
+        try {
+            Instrumentation instrumentation = ByteBuddyAgent.install();
+
+            new AgentBuilder.Default()
+                    .type(named("com.sun.tools.javac.comp.Lower"))
```

Comment:
```suggestion
                    .type(is(Lower.class))
```

Unless that doesn't work for some reason?